components/raftstore/src/store/worker/region.rs

```diff
@@ -543,6 +543,10 @@ where
             }
         }
 
+        // Notify the snapshot manager that a snapshot has been received and
+        // applied, freeing up the associated resource in the concurrency
+        // limiter.
+        self.mgr.recv_snap_complete(region_id);
```

Comment:
Should we consider TTL?

Reply:
You meant extending the TTL to forever when we receive the snapshot (so that the associated resource won't expire while the snapshot is being applied)? I was thinking of making that optimization in a follow-up PR to make the code review easier. But I can try to see if it's small enough to fit into this one.

Reply:
I felt that setting the TTL to forever after receiving the snapshot seemed inelegant and confusing. I ended up with a different approach: a new function `SnapRecvConcurrencyLimiter::set_reserved_capacity` is added to be able to mark some capacity as unavailable. Then, we maintain the invariant that the reserved capacity is equal to the number of pending applies. 

Now, `snap_mgr.recv_snap_complete` can be called post-snapshot reception, which alignes with the names of the functions and the limiter.